### PR TITLE
feat(fmt): add ability to configure semicolons

### LINKED ISSUE #17009: [fmt] Option to remove semicolons (please)
hello friends I would greatly appreciate it if I could turn off semicolons when I use `deno fmt`;
thanks for your understanding;

### PR DESCRIPTION
**Pull request by bartlomieju**
Always to change behavior of `deno fmt` to use "ASI" setting for semicolons instead of always prefering them.

I don't want it and think it's a bad idea, but the community has asked for it ~11 times in the past 4 weeks, so I think it's better to allow it than always deny this option and tell people to use other formatter.

Closes #17009

### CODE REVIEW

File: cli/tools/fmt.rs [hunk 1]
```diff
@@ -563,6 +564,15 @@ fn resolve_fmt_options(
     });
   }
 
+  if let Some(semicolons) = &fmt_flags.semicolons {
+    options.semi_colons = Some(match semicolons.as_str() {
+      "always" => SemiColons::Always,
```
Comment (dsherret) [inline cli/tools/fmt.rs:569]: Breaking change.

Comment (bartlomieju) [reply cli/tools/fmt.rs:569]: Good catch, thanks!

**Review by dsherret — APPROVED** [review]
LGTM

**Review by dsherret — APPROVED** [review]
LGTM ☹

**Comment by TodePond** [comment]
❤️ this is a beautiful day ❤️

**Pull request merged**